database-plugins/src/main/java/io/cdap/plugin/DataSizeReporter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.plugin;
+
+/**
+ * Reports data size.
+ *
+ */
+public interface DataSizeReporter {
+  /**
+   * Get the total size in bytes of data written.
+   * @returns number of bytes written.
+   */
+  long getBytesWritten();
```

Comment:
Given this new approach, couldnt we do the same for bytes read? 

We probably shouldnt add it now but it might make sense to add the method to the interface with a TODO to add the counting later. What do you think?

Reply:
Makes sense, I'll do that in a new PR because this is release/6.3 cherry-pick PR for #1267.